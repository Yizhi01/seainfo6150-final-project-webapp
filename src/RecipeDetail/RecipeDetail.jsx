import React from "react";
import { isEmpty } from "lodash";
import Breadcrumb from "../Breadcrumb/Breadcrumb.jsx";
import styles from "./RecipeDetail.module.css";
import PropTypes from "prop-types";

const RecipeDetail = (props) => {
  
  const { recipeId, recipes } = props;

  if (!isEmpty(recipes) && !isEmpty(recipes[recipeId - 1])) {
    const recipe = recipes[recipeId - 1];
    const ingredients = recipe.ingredients.map((ing, index) => (
      <li key={index}>{ing}</li>
    ));
    const instructions = Object.values(recipe.instructions).map((val, index) => 
      {
        return (
          <li key={index}>
            <span>Step {index + 1}:</span>
            <p>{val}</p>
          </li>
        )
      }
    );
  
    return (
      <div className={styles.recipeContainer}>
        <Breadcrumb categoryId={recipe.categoryId} categoryName={recipe.categoryName}/>
        <div className={styles.recipeTop}>
          <img className={styles.image} src={recipe.image._url} alt={recipe.name} />
          <div className={styles.recipeDetail}>
            <h1>{recipe.name}</h1>
            <p className={styles.description}>{recipe.description}</p>
            <hr/>
            <div className={styles.recipeInfo}>
              <p><span>SERVERS</span>{recipe.serves}</p>
              <p><span>COOKS IN</span>{recipe.cookTime}</p>
              <p><span>DIFFICULTY</span>{recipe.difficulty}</p>
            </div>
            <hr />
          </div>
        </div>
        <div className={styles.recipeContent}>
          <div className={styles.ingredients}>
            <h2>ingredients:</h2>
            <ul>{ingredients}</ul>
          </div>
          <div className={styles.instructions}>
            <h2>instructions:</h2>
            <ul>{instructions}</ul>
          </div>
        </div>
      </div>
    );
  } else {
    return <div>You have no data!</div>
  }
};

RecipeDetail.propTypes = {
  recipeId: PropTypes.string.isRequired,
  recipes: PropTypes.array.isRequired
};

export default RecipeDetail;