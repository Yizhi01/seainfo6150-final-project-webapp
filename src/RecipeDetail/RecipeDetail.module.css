.recipeContainer {
  margin: 30px 100px;
  display: flex;
  flex-direction: column;
}

.recipeTop {
  padding: 40px 0;
  display: grid;
  grid-template-columns: 1fr 2fr;
}

h1 {
  font-family: 'Castoro', serif;
}

hr {
  border: 1px solid #bedbbb;
}

.image {
  width: 100%;
}

.recipeDetail {
  margin: 0 40px 0 60px;
  display: flex;
  flex-direction: column;
}

.description {
  padding: 15px 0;
}

.recipeInfo {
  padding: 15px 0 0;
}

.recipeInfo p {
  margin: 0 0 15px;
}

.recipeInfo span {
  margin: 0 15px 0 0;
  color: #a05344;
}

.recipeContent {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

ul {
  list-style: none;
}

h2 {
  text-transform: capitalize;
  font-family: 'Castoro', serif;
}

.ingredients {
  padding: 0 50px 0 0;
}

.ingredients li {
  line-height: 40px;
}

.instructions li {
  padding: 15px 0;
  line-height: 25px;
}

@media (max-width: 900px) {
  .recipeContainer {
    margin: 10px 20px;
  }
}

@media (max-width: 600px) {
  .recipeTop {
    grid-template-columns: 1fr;
  }

  .recipeContent {
    grid-template-columns: 1fr;
  }
}
