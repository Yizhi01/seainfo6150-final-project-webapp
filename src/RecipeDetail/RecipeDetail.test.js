import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import RecipeDetail from "./RecipeDetail.jsx";

const testlist = [
  {
    "categoryId": "5",
    "recipeId": "1",
    "categoryName": "drinks",
    "name": "Fruit & Yogurt Smoothie",
    "description": "This easy fruit smoothie with yogurt recipe calls for just three ingredients--yogurt, fruit juice and whatever frozen fruit you have on hand. Mix up your combinations from day to day for a healthy breakfast or snack you'll never get bored with.",
    "ingredients": [
      "1 cup 3/4 cup nonfat plain yogurt",
      "¼ cup 1/2 cup 100% pure fruit juice ",
      "1½ cups 1 1/2 cups (6 1/2 ounces) frozen fruit, such as blueberries, raspberries, pineapple or peaches"
    ],
    "instructions": {
      "1": "Puree yogurt with juice in a blender until smooth. With the motor running, add fruit through the hole in the lid and continue to puree until smooth."
    },
    "pubTime": "2020-11-01",
    "serves": "1",
    "cookTime": "10 mins",
    "difficulty": "easy",
    "image": {
      "_url": "https://source.unsplash.com/CpBBsda2eRI/750x1000",
      "_height": "1000",
      "_width": "750"
    }
  }
]

it("renders the RecipeDetail component correctly", () => {
  const { container } = render(
    <Router>
      <RecipeDetail recipeId="1" recipes={testlist} />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
