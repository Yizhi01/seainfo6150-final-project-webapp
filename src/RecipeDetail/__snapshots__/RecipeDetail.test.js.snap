// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipeDetail component correctly 1`] = `
<div>
  <div
    class="recipeContainer"
  >
    <div>
      <nav>
        <ol
          class="breadcrumbList"
        >
          <li>
            <a
              class="breadcrumbTitle"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="breadcrumbTitle"
              href="/categoryList"
            >
              Recipes
            </a>
          </li>
          <li>
            <a
              class="breadcrumbTitle"
              href="/category/5/drinks"
            >
              drinks
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="recipeTop"
    >
      <img
        alt="Fruit & Yogurt Smoothie"
        class="image"
        src="https://source.unsplash.com/CpBBsda2eRI/750x1000"
      />
      <div
        class="recipeDetail"
      >
        <h1>
          Fruit & Yogurt Smoothie
        </h1>
        <p
          class="description"
        >
          This easy fruit smoothie with yogurt recipe calls for just three ingredients--yogurt, fruit juice and whatever frozen fruit you have on hand. Mix up your combinations from day to day for a healthy breakfast or snack you'll never get bored with.
        </p>
        <hr />
        <div
          class="recipeInfo"
        >
          <p>
            <span>
              SERVERS
            </span>
            1
          </p>
          <p>
            <span>
              COOKS IN
            </span>
            10 mins
          </p>
          <p>
            <span>
              DIFFICULTY
            </span>
            easy
          </p>
        </div>
        <hr />
      </div>
    </div>
    <div
      class="recipeContent"
    >
      <div
        class="ingredients"
      >
        <h2>
          ingredients:
        </h2>
        <ul>
          <li>
            1 cup 3/4 cup nonfat plain yogurt
          </li>
          <li>
            ¼ cup 1/2 cup 100% pure fruit juice 
          </li>
          <li>
            1½ cups 1 1/2 cups (6 1/2 ounces) frozen fruit, such as blueberries, raspberries, pineapple or peaches
          </li>
        </ul>
      </div>
      <div
        class="instructions"
      >
        <h2>
          instructions:
        </h2>
        <ul>
          <li>
            <span>
              Step 
              1
              :
            </span>
            <p>
              Puree yogurt with juice in a blender until smooth. With the motor running, add fruit through the hole in the lid and continue to puree until smooth.
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
